adventure.py

```diff
@@ -1,267 +1,514 @@
+        elif motion == "s":
+            mob_collision_s = detect_mob_collision("y", -1)
+            for i in range(0, player.speed):
+                if player_pos[y] - 1 > dim - 1 or player_pos[y] - 1 < 1:
+                    print("You cannot leave the map!")
+                    sleep(0.5)
+                elif detect_collision("y", -1)[0] == True:
+                    print("Collision detected:")
+                    print("You cannot traverse a {}".format(detect_collision("y", -1)[1]))
+                    sleep(0.5)
+                elif mob_collision_s[0] and len(mob_collision_s) >= 2:
+                    print("You attacked a {0}!\n{0}'s health is now {1}!".format(mob_collision_s[1], mob_collision_s[2]))
+                    sleep(0.5)
+                else:
+                    reset_pos()
+                    del stored_tile[0]
+                    player_pos[y] -= 1
+                    stored_tile.append(world.char(player_pos[x], player_pos[y]))
+        elif motion == "a":
+            mob_collision_a = detect_mob_collision("x", -1)
+            for i in range(0, player.speed):
+                if player_pos[x] - 1 > dim - 1 or player_pos[x] - 1 < 2:
+                    print("You cannot leave the map!")
+                    sleep(0.5)
+                elif detect_collision("x", -1)[0] == True:
+                    print("Collision detected:")
+                    print("You cannot traverse a {}".format(detect_collision("x", -1)[1]))
+                    sleep(0.5)
+                elif mob_collision_a[0] and len(mob_collision_a) >= 2:
+                    print("You attacked a {0}!\n{0}'s health is now {1}!".format(mob_collision_a[1], mob_collision_a[2]))
+                    sleep(0.5)
+                else:
+                    reset_pos()
+                    del stored_tile[0]
+                    player_pos[x] -= 1
+                    stored_tile.append(world.char(player_pos[x], player_pos[y]))
+        elif motion == "d":
+            mob_collision_d = detect_mob_collision("x", 1)
+            for i in range(0, player.speed):
+                if player_pos[x] + 1 > dim - 1 or player_pos[x] + 1 < 2:
+                    print("You cannot leave the map!")
+                    sleep(0.5)
+                elif detect_collision("x", 1)[0] == True:
+                    print("Collision detected:")
+                    print("You cannot traverse a {}".format(detect_collision("x", 1)[1]))
+                    sleep(0.5)
+                elif mob_collision_d[0] and len(mob_collision_d) >= 2:
+                    print("You attacked a {0}!\n{0}'s health is now {1}!".format(mob_collision_d[1], mob_collision_d[2]))
+                    sleep(0.5)
+                else:
+                    reset_pos()
+                    del stored_tile[0]
+                    player_pos[x] += 1
+                    stored_tile.append(world.char(player_pos[x], player_pos[y]))
+
+
+    def randomly_locate_player():
```

Comment:
Player should return to origin upon death, not be randomly located-- randomly locating could provide them with an advantage

Reply:
Addressed. See commit `962e6ce`.